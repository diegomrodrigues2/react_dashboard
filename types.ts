

import React from 'react';
import { SalesData as SalesDataType } from './components/Dashboard.tsx'; // Import to resolve name conflict


export interface ContaGestora {
  id: string;
  Conta: string | number; // Can be number
  NomeCliente: string;
  NomeClienteBTB?: string;
  SubManager?: string;
  TIPO?: string;
}

export interface ContaAssessor {
  id:string;
  AssessorSINACOROriginal: string;
  AssessorSINACOR: string;
  Assessor: string;
}

export interface CbioBtc {
  id: string;
  Historico: string; 
  Broker: string;
  ContaTipoB: string; // 'Y'/'N'
  Segmento: string;
  CommodityProduto: string;
  Mercado: string;
}

export interface SplitExcecao {
  id: string;
  DataAlteracao: string; // YYYY-MM-DD for date type
  Conta: string | number; // Can be number
  Segmento: string;
  Produto: string;
  Assessor: string;
  Proporcao: string | number; // e.g., "100" for 100% if number type
  AssessorSinacor: string;
  SegmentoCliente: string;
}

export interface BrokerCode {
  id: string;
  Type: string;
  OTCBrokerCode: string;
  FUTBrokerCode: string;
  Commission: string | number; // number if type 'number'
  ShareCode: string;
  BCTeam: string;
  TeamNickname: string;
  PoolName: string;
  Department: string;
  Entity: string;
  Region: string;
  Office: string;
  Head: string;
  ChangedOrCreatedOn: string; // YYYY-MM-DD for date type
  Status: string;
  Obs?: string;
}

export interface SalesData {
  id: string;
  mes: string;
  regiao: string;
  categoria: string;
  vendas: number;
  lucro: number;
  clientes: number;
  coordinates?: [number, number];
}

export type DataItem = ContaGestora | ContaAssessor | CbioBtc | SplitExcecao | BrokerCode | SalesData;

/** Utility type to extract string keys from a record */
export type DataKey<T extends object> = Extract<keyof T, string>;

/** Utility type for retrieving the value type of a key from a record */
export type DataValue<T extends object, K extends DataKey<T>> = T[K];

export interface ColumnConfig<T extends object | ActivityLog> {
  header: string;
  accessor: DataKey<T>;
  render?: (value: DataValue<T, DataKey<T>>, row: T) => React.ReactNode;
}

export interface FieldConfig<T extends object = DataItem> {
  name: DataKey<T>;
  label: string;
  type: 'text' | 'number' | 'select' | 'date';
  options?: string[];
  required?: boolean;
}

export interface MenuItem<T extends DataItem> {
  id: string;
  label: string;
  icon: React.FC<React.SVGProps<SVGSVGElement>>;
  columns?: ColumnConfig<T>[];
  fields?: FieldConfig<T>[];
  dataKey?: keyof AppData;
  getRowClass?: (item: T) => string;
  hidden?: boolean;
}

export interface AppData {
  contaGestora: ContaGestora[];
  contaAssessor: ContaAssessor[];
  cbioBtc: CbioBtc[];
  splitExcecao: SplitExcecao[];
  brokerCodes: BrokerCode[];
}

// User Profile Types
export type UserRole = 'admin' | 'editor' | 'viewer';

export interface UserData {
  id: string; // Typically the username
  name: string; // Display name, e.g., "Admin User"
  role: UserRole;
}

export interface ActivityLog {
  id: string;
  timestamp: string; 
  action: string;
  details: string;
  targetTable?: string; 
}

// Dashboard Types
export interface KpiData {
    totalVendas: number;
    totalLucro: number;
    totalClientes: number;
    ticketMedio: number;
}

export type ChartComponentType = 'KpiCard' | 'LineChart' | 'BarChart' | 'PieChart' | 'DataTable' | 'ScatterChart' | 'AreaChart' | 'ComboChart' | 'WaterfallChart' | 'FunnelChart' | 'TreemapChart' | 'Matrix' | 'BubbleMap' | 'ChoroplethMap' | 'HeatmapChart';

// Configuration for individual widgets
export interface KpiConfig {
    valueKey: keyof KpiData;
    description: string;
    formatter: (value: number) => string;
}

export interface DataTableConfig {
    dataKey: 'detailedData';
    columns: ColumnConfig<SalesDataType>[];
}

// New Dynamic Chart Configuration Types
export type AggregationType = 'SUM' | 'COUNT' | 'AVERAGE' | 'NONE';

export interface ChartWell<T extends object = Record<string, unknown>> {
    dataKey: DataKey<T> | 'category' | 'value' | 'stage' | 'name' | 'size' | 'colorMetric';
    label?: string;
    aggregation?: AggregationType;
    formatter?: (value: number) => string;
}

export interface DynamicChartConfig<T extends object = Record<string, unknown>> {
    sourceDataKey: 'detailedData' | 'waterfallData' | 'funnelData';
    chartType: 'Bar' | 'Line' | 'Pie' | 'Scatter' | 'Area' | 'Combo' | 'Waterfall' | 'Funnel' | 'Treemap';

    // Wells (data fields)
    category?: ChartWell<T>; // For Pie: slices. For Bar/Line: axis categories. For Scatter: 'Details' for each point. For Funnel/Treemap: category name.
    value?: ChartWell<T>;    // For Pie: slice values. For Bar/Line: axis values. For Combo: Bar value. For Waterfall: Change value. For Funnel: stage value. For Treemap: node size.
    colorValue?: ChartWell<T>; // For Treemap color scale.
    lineValue?: ChartWell<T>; // For Combo chart's line value

    // Wells for Scatter charts
    x?: ChartWell<T>;
    y?: ChartWell<T>;
    size?: ChartWell<T>;     // Turns Scatter into Bubble chart
    playAxis?: ChartWell<T>; // Adds animation dimension

    legend?: ChartWell<T>;  // For splitting/coloring data in Bar, Line, Scatter

    // Config specific to chart visuals
    layout?: 'horizontal' | 'vertical'; // For Bar charts
    stackType?: 'none' | 'stacked' | '100%stacked'; // For Bar/Area charts

    // Future-proofing for more advanced features
    smallMultiples?: ChartWell<T>;
    tooltips?: ChartWell<T>[];
}

export interface MatrixConfig<T extends object = Record<string, unknown>> {
    sourceDataKey: 'detailedData';
    rows: ChartWell<T>[];
    columns: ChartWell<T>[];
    values: ChartWell<T>[];
    rowSubtotals?: boolean;
    columnSubtotals?: boolean;
}

export interface HeatmapConfig<T extends object = Record<string, unknown>> {
    sourceDataKey: 'detailedData';
    x: ChartWell<T>;
    y: ChartWell<T>;
    value: ChartWell<T>;
}

export interface BubbleMapConfig<T extends object = Record<string, unknown>> {
    sourceDataKey: 'detailedData';
    location: {
        coordinatesKey: 'coordinates';
        labelKey: DataKey<T>;
    };
    size: ChartWell<T>;
    color: ChartWell<T>;
}

export interface ChoroplethMapConfig<T extends object = Record<string, unknown>> {
    sourceDataKey: 'detailedData';
    geo: {
        dataKey: DataKey<T>;
        featureKey: string;
    };
    value: ChartWell<T>;
}

export type WidgetConfig<T extends object = Record<string, unknown>> =
    | KpiConfig
    | DynamicChartConfig<T>
    | DataTableConfig
    | MatrixConfig<T>
    | BubbleMapConfig<T>
    | ChoroplethMapConfig<T>
    | HeatmapConfig<T>;

export interface WidgetLayout {
    x: number;
    y: number;
    w: number;
    h: number;
}

export interface DashboardWidget {
    id: string;
    component: ChartComponentType;
    gridClass: string;
    title?: string;
    config: WidgetConfig;
    isLoading?: boolean;
    error?: string;
    layout?: WidgetLayout;
}

export interface DashboardConfig {
    filters: ('month' | 'region' | 'category')[];
    widgets: DashboardWidget[];
}