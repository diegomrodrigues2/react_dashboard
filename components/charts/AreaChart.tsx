
import React, { useMemo } from 'react';
import { AreaChart as RechartsAreaChart, Area, XAxis, YAxis, CartesianGrid, Tooltip, Legend, ResponsiveContainer, Label } from 'recharts';
import { DynamicChartConfig } from '../../types.ts';

const areaColors = ['#00A3E0', '#D9262E', '#64B5F6', '#FFCA28', '#4CAF50', '#FF7043', '#2196F3', '#005A9C'];

interface AreaChartProps {
  data: any[];
  title: string;
  config: DynamicChartConfig;
  gradientColors?: [string, string];
}

const currencyFormatter = (val: any) => {
    if (typeof val !== 'number') return val;
    return new Intl.NumberFormat('pt-BR', {
      style: 'currency',
      currency: 'BRL',
      notation: 'compact',
      compactDisplay: 'short'
    }).format(val);
};

const CustomTooltip = ({ active, payload, label, config }: any) => {
    if (active && payload && payload.length) {
        const isStacked = config.stackType === 'stacked';
        const total = isStacked ? payload.reduce((sum: number, entry: any) => sum + entry.value, 0) : null;
        return (
            <div className="bg-white/90 p-3 border border-gray-200 rounded-lg shadow-lg backdrop-blur-sm">
                <p className="font-semibold text-gray-700 mb-2">{label}</p>
                {payload.map((pld: any, index: number) => (
                    <div key={index} style={{ color: pld.stroke }} className="text-sm">
                        {`${pld.name}: ${currencyFormatter(pld.value)}`}
                    </div>
                ))}
                {isStacked && total !== null && (
                     <div className="pt-2 mt-2 border-t border-gray-200 font-bold text-gray-800 text-sm">
                        {`Total: ${currencyFormatter(total)}`}
                    </div>
                )}
            </div>
        );
    }
    return null;
};


const AreaChart: React.FC<AreaChartProps> = ({ data, title, config, gradientColors }) => {
  const { category, value, legend, stackType = 'none' } = config;

  const areaDataKeys = useMemo(() => {
    if (!legend && value) {
      return [{ key: value.dataKey, name: value.label || String(value.dataKey) }];
    }
    if (!legend || !category || !data || data.length === 0) return [];
    const firstRow = data[0];
    return Object.keys(firstRow)
      .filter(key => key !== category.dataKey)
      .map(key => ({ key, name: key }));
  }, [data, legend, category, value]);
  
  const isStacked = stackType === 'stacked' || stackType === '100%stacked';
  const is100Percent = stackType === '100%stacked';
  
  const percentFormatter = (tick: number) => `${(tick * 100).toFixed(0)}%`;

  return (
    <div className="bg-white p-6 rounded-lg shadow-lg h-96 flex flex-col">
      <h3 className="text-lg font-semibold text-gray-800 mb-4 text-center">{title}</h3>
      <div className="flex-grow">
        <ResponsiveContainer width="100%" height="100%">
            <RechartsAreaChart 
              data={data} 
              margin={{ top: 5, right: 30, left: 20, bottom: 20 }}
              stackOffset={is100Percent ? 'expand' : undefined}
            >
              <defs>
                {gradientColors ? (
                  <linearGradient id="customGradient" x1="0" y1="0" x2="0" y2="1">
                    <stop offset="5%" stopColor={gradientColors[0]} stopOpacity={isStacked ? 0.8 : 0.6} />
                    <stop offset="95%" stopColor={gradientColors[1] ?? gradientColors[0]} stopOpacity={isStacked ? 0.2 : 0} />
                  </linearGradient>
                ) : (
                  areaDataKeys.map((area, index) => (
                    <linearGradient key={`color-${area.key}`} id={`color-${area.key}`} x1="0" y1="0" x2="0" y2="1">
                      <stop offset="5%" stopColor={areaColors[index % areaColors.length]} stopOpacity={isStacked ? 0.8 : 0.6} />
                      <stop offset="95%" stopColor={areaColors[index % areaColors.length]} stopOpacity={isStacked ? 0.2 : 0} />
                    </linearGradient>
                  ))
                )}
              </defs>
              <CartesianGrid strokeDasharray="3 3" vertical={false} />
              <XAxis dataKey={category?.dataKey} stroke="#6B7280" tick={{ fontSize: 12 }}>
                 <Label value={category?.label} offset={-15} position="insideBottom" fill="#6B7280" fontSize={14} />
              </XAxis>
              <YAxis 
                stroke="#6B7280" 
                tickFormatter={is100Percent ? percentFormatter : currencyFormatter} 
                tick={{ fontSize: 12 }}
                domain={is100Percent ? [0, 1] : undefined}
              >
                <Label value={value?.label} angle={-90} offset={-5} position="insideLeft" style={{ textAnchor: 'middle' }} fill="#6B7280" fontSize={14} />
              </YAxis>
              <Tooltip
                  content={<CustomTooltip config={config} />}
                  contentStyle={{
                      backgroundColor: 'rgba(255, 255, 255, 0.9)',
                      border: '1px solid #E5E7EB',
                      borderRadius: '0.5rem',
                  }}
              />
              <Legend wrapperStyle={{fontSize: "12px", paddingTop: "10px"}}/>
              {areaDataKeys.map((area, index) => (
                  <Area
                      key={area.key}
                      type="monotone"
                      dataKey={area.key}
                      name={area.name}
                      stroke={areaColors[index % areaColors.length]}
                      strokeWidth={2}
                      fillOpacity={1}
                      fill={gradientColors ? 'url(#customGradient)' : `url(#color-${area.key})`}
                      stackId={isStacked ? '1' : undefined}
                  />
              ))}
            </RechartsAreaChart>
        </ResponsiveContainer>
      </div>
    </div>
  );
};

export default AreaChart;
