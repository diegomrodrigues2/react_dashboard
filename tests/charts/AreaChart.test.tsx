import { render, screen, fireEvent } from '../setup.ts';
import { describe, it, expect, beforeAll } from 'vitest';
import AreaChart from '../../components/charts/AreaChart.tsx';
import { DynamicChartConfig } from '../../types.ts';

describe('AreaChart', () => {
  beforeAll(() => {
    // mock ResizeObserver used by Recharts' ResponsiveContainer
    (global as any).ResizeObserver = class {
      private callback: any;
      constructor(cb: any) {
        this.callback = cb;
      }
      observe() {
        this.callback([{ contentRect: { width: 400, height: 300 } }]);
      }
      unobserve() {}
      disconnect() {}
    };
    Object.defineProperty(HTMLElement.prototype, 'clientWidth', {
      configurable: true,
      value: 400,
    });
    Object.defineProperty(HTMLElement.prototype, 'clientHeight', {
      configurable: true,
      value: 300,
    });
    Object.defineProperty(HTMLElement.prototype, 'offsetWidth', {
      configurable: true,
      value: 400,
    });
    Object.defineProperty(HTMLElement.prototype, 'offsetHeight', {
      configurable: true,
      value: 300,
    });
    Object.defineProperty(HTMLElement.prototype, 'getBoundingClientRect', {
      configurable: true,
      value: function () {
        return {
          width: parseInt((this as HTMLElement).style.width || '400', 10),
          height: parseInt((this as HTMLElement).style.height || '300', 10),
          top: 0,
          left: 0,
          bottom: 0,
          right: 0,
        } as DOMRect;
      },
    });
  });
  const data = [
    { month: 'Jan', apples: 100, bananas: 200 },
    { month: 'Feb', apples: 150, bananas: 250 }
  ];

  const config: DynamicChartConfig = {
    sourceDataKey: 'detailedData',
    chartType: 'Area',
    category: { dataKey: 'month', label: 'Month' },
    legend: { dataKey: 'fruit' },
    value: { dataKey: 'apples', aggregation: 'SUM', label: 'Value' },
  } as any;

  it('snapshot renders', () => {
    const { container } = render(
      <div style={{ width: 400, height: 300 }}>
        <AreaChart data={data} title="Fruits" config={config} />
      </div>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('allows legend items to be reordered', () => {
    render(
      <div style={{ width: 400, height: 300 }}>
        <AreaChart data={data} title="Fruits" config={config} />
      </div>
    );
    const items = screen.getAllByTestId('legend-item');
    expect(items[0].textContent).toContain('apples');

    const dataTransfer: any = {
      data: {},
      setData(key: string, value: string) { this.data[key] = value; },
      getData(key: string) { return this.data[key]; }
    };

    fireEvent.dragStart(items[0], { dataTransfer });
    fireEvent.dragOver(items[1], { dataTransfer });
    fireEvent.drop(items[1], { dataTransfer });

    const reordered = screen.getAllByTestId('legend-item');
    expect(reordered[0].textContent).toContain('bananas');
  });
});
