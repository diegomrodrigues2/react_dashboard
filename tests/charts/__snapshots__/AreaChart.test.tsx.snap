// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AreaChart > snapshot renders 1`] = `
<div
  style="width: 400px; height: 300px;"
>
  <div
    class="bg-white p-6 rounded-lg shadow-lg h-96 flex flex-col"
  >
    <div
      class="flex items-center justify-between mb-4"
    >
      <h3
        class="text-lg font-semibold text-gray-800 text-center flex-1"
      >
        Fruits
      </h3>
    </div>
    <div
      class="flex-grow"
    >
      <div
        class="recharts-responsive-container"
        style="width: 100%; height: 100%; min-width: 0;"
      >
        <div
          class="recharts-wrapper"
          style="position: relative; cursor: default; width: 100%; height: 100%; max-height: 300px; max-width: 400px;"
        >
          <svg
            class="recharts-surface"
            height="300"
            style="width: 100%; height: 100%;"
            viewBox="0 0 400 300"
            width="400"
          >
            <title />
            <desc />
            <defs>
              <clippath
                id="recharts1-clip"
              >
                <rect
                  height="245"
                  width="290"
                  x="80"
                  y="5"
                />
              </clippath>
            </defs>
            <defs>
              <lineargradient
                id="color-apples"
                x1="0"
                x2="0"
                y1="0"
                y2="1"
              >
                <stop
                  offset="5%"
                  stop-color="#00A3E0"
                  stop-opacity="0.6"
                />
                <stop
                  offset="95%"
                  stop-color="#00A3E0"
                  stop-opacity="0"
                />
              </lineargradient>
              <lineargradient
                id="color-bananas"
                x1="0"
                x2="0"
                y1="0"
                y2="1"
              >
                <stop
                  offset="5%"
                  stop-color="#D9262E"
                  stop-opacity="0.6"
                />
                <stop
                  offset="95%"
                  stop-color="#D9262E"
                  stop-opacity="0"
                />
              </lineargradient>
            </defs>
            <g
              class="recharts-cartesian-grid"
            >
              <g
                class="recharts-cartesian-grid-horizontal"
              >
                <line
                  fill="none"
                  height="245"
                  stroke="#ccc"
                  stroke-dasharray="3 3"
                  width="290"
                  x="80"
                  x1="80"
                  x2="370"
                  y="5"
                  y1="250"
                  y2="250"
                />
                <line
                  fill="none"
                  height="245"
                  stroke="#ccc"
                  stroke-dasharray="3 3"
                  width="290"
                  x="80"
                  x1="80"
                  x2="370"
                  y="5"
                  y1="188.75"
                  y2="188.75"
                />
                <line
                  fill="none"
                  height="245"
                  stroke="#ccc"
                  stroke-dasharray="3 3"
                  width="290"
                  x="80"
                  x1="80"
                  x2="370"
                  y="5"
                  y1="127.5"
                  y2="127.5"
                />
                <line
                  fill="none"
                  height="245"
                  stroke="#ccc"
                  stroke-dasharray="3 3"
                  width="290"
                  x="80"
                  x1="80"
                  x2="370"
                  y="5"
                  y1="66.25"
                  y2="66.25"
                />
                <line
                  fill="none"
                  height="245"
                  stroke="#ccc"
                  stroke-dasharray="3 3"
                  width="290"
                  x="80"
                  x1="80"
                  x2="370"
                  y="5"
                  y1="5"
                  y2="5"
                />
              </g>
            </g>
            <g
              class="recharts-layer recharts-cartesian-axis recharts-xAxis xAxis"
            >
              <line
                class="recharts-cartesian-axis-line"
                fill="none"
                height="30"
                orientation="bottom"
                stroke="#6B7280"
                width="290"
                x="80"
                x1="80"
                x2="370"
                y="250"
                y1="250"
                y2="250"
              />
              <g
                class="recharts-cartesian-axis-ticks"
              >
                <g
                  class="recharts-layer recharts-cartesian-axis-tick"
                >
                  <line
                    class="recharts-cartesian-axis-tick-line"
                    fill="none"
                    height="30"
                    orientation="bottom"
                    stroke="#6B7280"
                    width="290"
                    x="80"
                    x1="80"
                    x2="80"
                    y="250"
                    y1="256"
                    y2="250"
                  />
                  <text
                    class="recharts-text recharts-cartesian-axis-tick-value"
                    fill="#6B7280"
                    font-size="12"
                    height="30"
                    orientation="bottom"
                    stroke="none"
                    text-anchor="middle"
                    width="290"
                    x="80"
                    y="258"
                  >
                    <tspan
                      dy="0.71em"
                      x="80"
                    >
                      Jan
                    </tspan>
                  </text>
                </g>
                <g
                  class="recharts-layer recharts-cartesian-axis-tick"
                >
                  <line
                    class="recharts-cartesian-axis-tick-line"
                    fill="none"
                    height="30"
                    orientation="bottom"
                    stroke="#6B7280"
                    width="290"
                    x="80"
                    x1="370"
                    x2="370"
                    y="250"
                    y1="256"
                    y2="250"
                  />
                  <text
                    class="recharts-text recharts-cartesian-axis-tick-value"
                    fill="#6B7280"
                    font-size="12"
                    height="30"
                    orientation="bottom"
                    stroke="none"
                    text-anchor="middle"
                    width="290"
                    x="370"
                    y="258"
                  >
                    <tspan
                      dy="0.71em"
                      x="370"
                    >
                      Feb
                    </tspan>
                  </text>
                </g>
              </g>
              <text
                class="recharts-text recharts-label"
                fill="#6B7280"
                font-size="14"
                offset="-15"
                text-anchor="middle"
                x="225"
                y="295"
              >
                <tspan
                  dy="0em"
                  x="225"
                >
                  Month
                </tspan>
              </text>
            </g>
            <g
              class="recharts-layer recharts-cartesian-axis recharts-yAxis yAxis"
            >
              <line
                class="recharts-cartesian-axis-line"
                fill="none"
                height="245"
                orientation="left"
                stroke="#6B7280"
                width="60"
                x="20"
                x1="80"
                x2="80"
                y="5"
                y1="5"
                y2="250"
              />
              <g
                class="recharts-cartesian-axis-ticks"
              >
                <g
                  class="recharts-layer recharts-cartesian-axis-tick"
                >
                  <line
                    class="recharts-cartesian-axis-tick-line"
                    fill="none"
                    height="245"
                    orientation="left"
                    stroke="#6B7280"
                    width="60"
                    x="20"
                    x1="74"
                    x2="80"
                    y="5"
                    y1="250"
                    y2="250"
                  />
                  <text
                    class="recharts-text recharts-cartesian-axis-tick-value"
                    fill="#6B7280"
                    font-size="12"
                    height="245"
                    orientation="left"
                    stroke="none"
                    text-anchor="end"
                    width="60"
                    x="72"
                    y="250"
                  >
                    <tspan
                      dy="0.355em"
                      x="72"
                    >
                      R$Â 0
                    </tspan>
                  </text>
                </g>
                <g
                  class="recharts-layer recharts-cartesian-axis-tick"
                >
                  <line
                    class="recharts-cartesian-axis-tick-line"
                    fill="none"
                    height="245"
                    orientation="left"
                    stroke="#6B7280"
                    width="60"
                    x="20"
                    x1="74"
                    x2="80"
                    y="5"
                    y1="188.75"
                    y2="188.75"
                  />
                  <text
                    class="recharts-text recharts-cartesian-axis-tick-value"
                    fill="#6B7280"
                    font-size="12"
                    height="245"
                    orientation="left"
                    stroke="none"
                    text-anchor="end"
                    width="60"
                    x="72"
                    y="188.75"
                  >
                    <tspan
                      dy="0.355em"
                      x="72"
                    >
                      R$Â 65
                    </tspan>
                  </text>
                </g>
                <g
                  class="recharts-layer recharts-cartesian-axis-tick"
                >
                  <line
                    class="recharts-cartesian-axis-tick-line"
                    fill="none"
                    height="245"
                    orientation="left"
                    stroke="#6B7280"
                    width="60"
                    x="20"
                    x1="74"
                    x2="80"
                    y="5"
                    y1="127.5"
                    y2="127.5"
                  />
                  <text
                    class="recharts-text recharts-cartesian-axis-tick-value"
                    fill="#6B7280"
                    font-size="12"
                    height="245"
                    orientation="left"
                    stroke="none"
                    text-anchor="end"
                    width="60"
                    x="72"
                    y="127.5"
                  >
                    <tspan
                      dy="0.355em"
                      x="72"
                    >
                      R$Â 130
                    </tspan>
                  </text>
                </g>
                <g
                  class="recharts-layer recharts-cartesian-axis-tick"
                >
                  <line
                    class="recharts-cartesian-axis-tick-line"
                    fill="none"
                    height="245"
                    orientation="left"
                    stroke="#6B7280"
                    width="60"
                    x="20"
                    x1="74"
                    x2="80"
                    y="5"
                    y1="66.25"
                    y2="66.25"
                  />
                  <text
                    class="recharts-text recharts-cartesian-axis-tick-value"
                    fill="#6B7280"
                    font-size="12"
                    height="245"
                    orientation="left"
                    stroke="none"
                    text-anchor="end"
                    width="60"
                    x="72"
                    y="66.25"
                  >
                    <tspan
                      dy="0.355em"
                      x="72"
                    >
                      R$Â 195
                    </tspan>
                  </text>
                </g>
                <g
                  class="recharts-layer recharts-cartesian-axis-tick"
                >
                  <line
                    class="recharts-cartesian-axis-tick-line"
                    fill="none"
                    height="245"
                    orientation="left"
                    stroke="#6B7280"
                    width="60"
                    x="20"
                    x1="74"
                    x2="80"
                    y="5"
                    y1="5"
                    y2="5"
                  />
                  <text
                    class="recharts-text recharts-cartesian-axis-tick-value"
                    fill="#6B7280"
                    font-size="12"
                    height="245"
                    orientation="left"
                    stroke="none"
                    text-anchor="end"
                    width="60"
                    x="72"
                    y="5"
                  >
                    <tspan
                      dy="0.355em"
                      x="72"
                    >
                      R$Â 260
                    </tspan>
                  </text>
                </g>
              </g>
              <text
                class="recharts-text recharts-label"
                fill="#6B7280"
                font-size="14"
                offset="-5"
                style="text-anchor: middle;"
                text-anchor="start"
                transform="rotate(-90, 15, 127.5)"
                x="15"
                y="127.5"
              >
                <tspan
                  dy="0.355em"
                  x="15"
                >
                  Value
                </tspan>
              </text>
            </g>
            <g
              class="recharts-layer recharts-area"
            >
              <g
                class="recharts-layer"
              >
                <defs>
                  <clippath
                    id="animationClipPath-recharts-area-4"
                  >
                    <rect
                      height="252"
                      width="0"
                      x="80"
                      y="0"
                    />
                  </clippath>
                </defs>
                <g
                  class="recharts-layer"
                  clip-path="url(#animationClipPath-recharts-area-4)"
                >
                  <g
                    class="recharts-layer"
                  >
                    <path
                      class="recharts-curve recharts-area-area"
                      d="M80,155.7692307692308L370,108.65384615384617L370,250L80,250Z"
                      fill="url(#color-apples)"
                      fill-opacity="1"
                      height="245"
                      name="apples"
                      stroke="none"
                      stroke-width="2"
                      width="290"
                    />
                    <path
                      class="recharts-curve recharts-area-curve"
                      d="M80,155.7692307692308L370,108.65384615384617"
                      fill="none"
                      fill-opacity="1"
                      height="245"
                      name="apples"
                      stroke="#00A3E0"
                      stroke-width="2"
                      width="290"
                    />
                  </g>
                </g>
              </g>
            </g>
            <g
              class="recharts-layer recharts-area"
            >
              <g
                class="recharts-layer"
              >
                <defs>
                  <clippath
                    id="animationClipPath-recharts-area-5"
                  >
                    <rect
                      height="252"
                      width="0"
                      x="80"
                      y="0"
                    />
                  </clippath>
                </defs>
                <g
                  class="recharts-layer"
                  clip-path="url(#animationClipPath-recharts-area-5)"
                >
                  <g
                    class="recharts-layer"
                  >
                    <path
                      class="recharts-curve recharts-area-area"
                      d="M80,61.538461538461526L370,14.423076923076916L370,250L80,250Z"
                      fill="url(#color-bananas)"
                      fill-opacity="1"
                      height="245"
                      name="bananas"
                      stroke="none"
                      stroke-width="2"
                      width="290"
                    />
                    <path
                      class="recharts-curve recharts-area-curve"
                      d="M80,61.538461538461526L370,14.423076923076916"
                      fill="none"
                      fill-opacity="1"
                      height="245"
                      name="bananas"
                      stroke="#D9262E"
                      stroke-width="2"
                      width="290"
                    />
                  </g>
                </g>
              </g>
            </g>
          </svg>
          <div
            class="recharts-legend-wrapper"
            style="position: absolute; width: 350px; height: auto; left: 20px; bottom: 20px;"
          >
            <ul
              class="flex flex-wrap"
              style="list-style: none; margin: 0px; padding: 0px;"
            >
              <li
                class="flex items-center mr-4 cursor-move"
                data-testid="legend-item"
                draggable="true"
              >
                <span
                  class="mr-1"
                >
                  â°
                </span>
                <svg
                  class="mr-1"
                  height="10"
                  viewBox="0 0 32 32"
                  width="10"
                >
                  <rect
                    fill="#00A3E0"
                    height="32"
                    width="32"
                  />
                </svg>
                <span>
                  apples
                </span>
              </li>
              <li
                class="flex items-center mr-4 cursor-move"
                data-testid="legend-item"
                draggable="true"
              >
                <span
                  class="mr-1"
                >
                  â°
                </span>
                <svg
                  class="mr-1"
                  height="10"
                  viewBox="0 0 32 32"
                  width="10"
                >
                  <rect
                    fill="#D9262E"
                    height="32"
                    width="32"
                  />
                </svg>
                <span>
                  bananas
                </span>
              </li>
            </ul>
          </div>
          <div
            class="recharts-tooltip-wrapper"
            style="visibility: hidden; pointer-events: none; position: absolute; top: 0px; left: 0px;"
            tabindex="-1"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
